### PR TITLE
Update WC Rest API to 1.0.10-pl-1 for inclusion in 4.3.2

### PR DESCRIPTION
**Pull request by vedanshujain**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates WC REST API to include changes for WP 5.5 compatibility.

### Changelog

* Enhancement - Compatibility fixes for WordPress 5.5 woocommerce/woocommerce-rest-api#232

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enhancement - Update REST API to 1.0.10-pl-1

### CODE REVIEW

**Comment by rrennick** [comment]
@vedanshujain Is this ready for review?

**Comment by vedanshujain** [comment]
Note that failing nightly test will pass after #27236 is merged (checked locally)

**Comment by vedanshujain** [comment]
@rrennick ready for review now. (apologies for not marking it on friday)

**Review by rrennick — APPROVED** [review]
@vedanshujain Worked great for me.

File: composer.lock [hunk 1]
```diff
@@ -44,16 +44,16 @@
         },
         {
             "name": "automattic/jetpack-constants",
-            "version": "v1.3.0",
+            "version": "v1.4.0",
```
Comment (peterfabian) [inline composer.lock:47]: Can you please update the composer.lock only for the REST API package?

**Review by peterfabian — CHANGES_REQUESTED** [review]
Works fine, just need to update the single package in the .lock file please.

File: composer.lock (hunk 1 above)
Comment (vedanshujain) [reply composer.lock:47]: Taking a look

**Review by peterfabian — APPROVED** [review]
LGTM!

**Pull request merged**